src/module/rules/rule-element/special-resource.ts

```diff
@@ -0,0 +1,198 @@
+import { ActorType, CharacterPF2e } from "@actor";
+import { CharacterResources } from "@actor/character/data.ts";
+import { PhysicalItemPF2e } from "@item";
+import { PhysicalItemSource } from "@item/base/data/index.ts";
+import { sluggify } from "@util";
+import { NumberField, StringField } from "types/foundry/common/data/fields.js";
+import { createBatchRuleElementUpdate } from "../helpers.ts";
+import { RuleElementOptions, RuleElementPF2e } from "./base.ts";
+import { ResolvableValueField, RuleElementSchema, RuleElementSource } from "./data.ts";
+
+const INVALID_RESOURCES: (keyof CharacterResources)[] = [
+    "crafting",
+    "focus",
+    "heroPoints",
+    "investiture",
+    "infusedReagents",
+    "resolve",
+];
+
+class SpecialResourceRuleElement extends RuleElementPF2e<SpecialResourceSchema> {
+    protected static override validActorTypes: ActorType[] = ["character", "npc"];
```

Comment:
OK for now, but we should look at expanding this as people request it.

Reply:
This was originally character and I extended it to npc, but to extend further we would need to ensure that the resource object exists at least at the creature level.

Reply:
I ended up removing NPC, as we're not using it for mythic points. There was a 50/50 chance when I wrote this PR but now its vanishingly small. This PR doesn't include any UI for the NPC resource either. We can add npc/familiars as people request them.